mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -64,6 +67,51 @@ struct AttentionRewritePattern : public OpConversionPattern<AttentionOp> {
   LogicalResult matchAndRewrite(AttentionOp op, AttentionOpAdaptor adaptor,
                                 ConversionPatternRewriter &b) const override;
 };
+
+static Type getSmallestType(Type type1, Type type2) {
+  return (type1.getIntOrFloatBitWidth() > type2.getIntOrFloatBitWidth())
+             ? type2
+             : type1;
+}
+
+static Type deduceAccumulatorElementType(Type elementTypeA, Type elementTypeB,
+                                         Type elementTypeC,
+                                         OpBuilder &builder) {
+  // Determine the type used on VGPR to act as accumulator.
+  // f32: f32.
+  // f16, bf16: f32 to prevent overflow from happening.
+  // i16 : i16.
+  // fp8 (any combo) : f32.
+  // i8: i32, since we have an i32 output
+  auto type = getSmallestType(elementTypeA, elementTypeB);
+  if (type.isa<FloatType>() && type.getIntOrFloatBitWidth() < 32) {
+    return builder.getF32Type();
+  } else if (type.isInteger(8)) {
+    return builder.getI32Type();
+  }
+  return elementTypeC;
+}
+
+static auto getAccumulator(Value a, Value b, Value c, OpBuilder &builder,
```

Comment:
Nit: Locations don't need to be passed by reference.

Reply:
yes, makes sense